adagrams/game.py

```diff
@@ -1,11 +1,145 @@
+        # use random.randint to pick a letter by index
+        index = random.randint(0, len(letters) - 1)        # generates a random index between 0 and len(letters) - 1
+        return letters[index]
+    
+    draw_letters = []
+
+    for i in range(10):    #select exactly 10 letters from dict
+        letter = pick_random_letter()
+        if letter is None:         # if None is returned indicating no more letters available, break the loop
+            break
+        draw_letters.append(letter)    # adding the randomly picked letters into the initial empty list
+        LETTER_POOL[letter] -= 1       # decrease the count of the drawn letter in the dict
+        if LETTER_POOL[letter] == 0:   # if the count of a letter reaches 0, removes the letter from the dict
+            del LETTER_POOL[letter] # remove the letter if its value reaches 0
+
+    return draw_letters
+
+result = draw_letters()
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # function that counts each letters that occured in the list of letters 
+    # this part of the code can be replaced by import counter
+    def count_letters(letters):
+        counts = {}
+        for letter in letters:
+            if letter in counts:
+                counts[letter] += 1
+            else:
+                counts[letter] = 1
+        return counts
+    
+    # convert word and letter_bank to lowercase to handle case sensitiveness
+    word = word.lower()
+    letter_bank = [letter.lower() for letter in letter_bank]
+    
+    # count the frequencies of the letters in word and letter_bank (word represents the user input words and letter_bank represent the letters that have been drawn)
+    word_counts = count_letters(word)
+    letter_bank_counts = count_letters(letter_bank)
+
+    # check to see if every letters in word is available in letter_bank
+    for letter, count in word_counts.items():
+        if letter_bank_counts.get(letter, 0) < count:
+            return False           # if the letter count in word is more than what we have in letter_bank
+    return True                    # if all the letters are available in letter_bank   
+
 
 def score_word(word):
-    pass
+    # score map, value of each letters
+    letter_scores = {
+        'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1,
+        'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+        'D': 2, 'G': 2,
+        'B': 3, 'C': 3, 'M': 3, 'P': 3,
+        'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+        'K': 5,
+        'J': 8, 'X': 8,
+        'Q': 10, 'Z': 10
+    }
+    # convert the word to uppercase to match the key in letter_score
+    word = word.upper()
+    # initialize base score
+    base_score = 0
+    for letter in word:
+        base_score += letter_scores.get(letter, 0)
+
+    # calculate the bonus points for the words od length 7 or more
+    if len(word) >= 7:
+        bonus_points = 8
+    else: 
+        bonus_points = 0
+
+    total_score = base_score + bonus_points
```

Comment:
As you get more comfortable with python, I would encourage you to keep the score to a single variable for the entirety of the function! It helps streamline things just a bit! This would replace the instinct to separate out the score into base and bonus and just keep it all in one place and add it up as the function progresses.